cyberdrop_dl/downloader/downloader.py

```diff
@@ -196,16 +196,24 @@ async def attempt_task_removal(self, media_item: MediaItem) -> None:
     @retry
     async def download(self, media_item: MediaItem) -> None:
         """Downloads the media item"""
-        if not await self.check_file_can_download(media_item):
-            await self.manager.progress_manager.download_progress.add_skipped()
+        can_download, reason = await self.check_file_can_download(media_item)
+        if not can_download:
+            if reason == 0:
+                await self.manager.progress_manager.download_progress.add_failed()
+            else:
+                await self.manager.progress_manager.download_progress.add_skipped()
```

Comment:
isn't this redundant? the exact same check is done just bellow, inside the try/except block, L211

Reply:
Pretty sure you're right... I didn't wanna mess too much with Jules' code that I don't have too much knowledge on 😅